src/main/java/uk/gov/justice/hmpps/prison/service/AgencyService.java

```diff
@@ -297,7 +297,10 @@ private List<Location> getAgencyLocationsOnDayAndPeriod(final String agencyId, @
 
         final var locations = agencyRepository.getAgencyLocationsBooked(agencyId, bookedOnDay, bookedOnPeriod);
         final var processedLocations = LocationProcessor.processLocations(locations, true);
-        return processedLocations.stream().sorted(LOCATION_DESCRIPTION_COMPARATOR).toList();
+        return processedLocations.stream()
+            .filter(location -> StringUtils.isNotBlank(location.getUserDescription()))
+            .sorted(LOCATION_DESCRIPTION_COMPARATOR)
+            .toList();
```

Comment:
This endpoint returns location ids with just the user description if there is one, in both the userDescription and description (not nullable) fields. If there isnt a user description, the issue is that the client gets a json error because description is missing. So the fix is to just omit these values. They are caused by in-cell bookings because cells have no user description. And it is ok for cell location not to be mentioned in the activities page which uses this.

Reply:
This feels like a bit of a hack. If the location has been used for an event then pretending it doesn't exist because it has a blank user description is surely wrong - it's being used after all? Better to change the OpenAPI specifiation to allow blank description and the client can ignore it if they really want to?

Reply:
Yes definitely, Ill be raising a ticket to overhaul the endpoint.